ivy/functional/frontends/numpy/creation_routines/building_matrices.py

```diff
@@ -36,3 +36,8 @@ def vander(x, N=None, increasing=False):
     elif ivy.is_bool_dtype or ivy.is_int_dtype(x):
         x = x.astype(ivy.int64)
     return ivy.vander(x, N=N, increasing=increasing)
+
+
+# diagflat
+def diagflat(v, k=0):
```

Comment:
please add the decorator `@to_ivy_arrays_and_back`

Reply:
changes done!